### ISSUE TITLE
Grid is not accessible using VoiceOver

### ISSUE THREAD

**Issue by backwardok** [bvaughn/react-virtualized#1199]
## Reporting a Bug

Steps to reproduce:
1. Turn on VoiceOver (VO) and open Safari
2. Open a Grid example (I used the [example in the documentation](https://bvaughn.github.io/react-virtualized/#/components/Grid))
3. Use standard VO navigation (ctrl + option + arrows) to get to the grid
4. Try to navigate into the grid either by using standard VO navigation

### What is the current behavior?

Grid is simply named as "grid" and VO announces that there are 0 rows and 0 columns. Trying to navigate into the grid using the default "step into" key combination (ctrl + option + shift + down arrow) doesn't do anything. 

### What is the expected behavior?

The grid should have a meaningful name (not just "grid"). There should be an accurate number of rows and columns. There should also be a way to get into the grid and navigate around to hear the different values in the grid.

This is achievable by wrapping each row in a container and using `role="row"` on that container. Each column header cell should have `role="columnheader"`. Each row header cell should have `role-"rowheader"`. Each data cell should have `role="gridcell"`.

If the data in the table is dynamically loaded, then `aria-colcount` and `aria-rowcount` should be set on the the container that has `role="grid"`.

Additionally, `aria-readonly="true"` is currently getting set for all cases, even if the elements aren't ever editable. This attribute should only be applied if there is a state when they will be editable. Otherwise, that attribute should just be omitted entirely.

For more detailed information on this, check out http://w3c.github.io/aria-practices/#grid.

### Which versions of React and react-virtualized, and which browser / OS are affected by this issue? Did this work in previous versions of react-virtualized?
|                   |          |
|-------------------|----------|
| Browser           |  Safari (using 11.1.2, but repros on 10.x as well)        |
| OS                | MacOS (tested with 10.13.6)         |
| React             | ^16.2.0 (whatever your documentation server is running, but verified also in 16.2.0)         |
| React DOM         | ^16.2.0 (whatever your documentation server is running, but verified also in 16.2.1)         |
| react-virtualized | Tested on 9.20.1 (and verified it exists in 9.18.3)         |

This also occurs on iOS using Safari running VoiceOver there as well.

**Comment by jsomsanith** [comment]
BTW I need that for my components too, and I'm interested in contributing, I started to fork and add the changes : https://github.com/jsomsanith/react-virtualized/tree/jsomsanith/fix/table_a11y
Tests missing.

Can any maintainer assign this issue to me ?

**Comment by jsomsanith** [comment]
@backwardok I tested my PR with VO + safari
* it still announce the number of rendered rows instead of aria-colcount value. This must be a VO issue.
* no (ctrl + option + shift + down arrow) action when we focus on div with `role=grid`, it doesn't even propose it. But the aria-label seem correct

```
<div role="grid" aria-label="from props" aria-colcount="2" aria-rowcount="1" /> // label and col/row count
	<div role="rowheader"> . // row --> rowheader
		<div role="columnheader" aria-sort="none"/> // sortable but not current
		<div role="columnheader"/>
	</div>
	<div role="rowgroup">
		<div role="row" aria-rowindex="1" > // aria-rowindex
			<div role="gridcell" aria-colindex="1" /> // aria-colindex
			<div role="gridcell" aria-colindex="2" />
		</div>
	</div>
</div>
```
